### DISCUSSION TITLE
Will `quarto preview` also do the works that `quarto render` do? Where is the output of `quarto preview`?

### DISCUSSION THREAD

**Discussion by 2catycm** [quarto-dev/quarto-cli#11588] [Q&A]
### Description

`quarto render` is very slow and renders every file every time. I have 20+ ipynb files in a project using quarto to render, and it take me about half an hour to see it finish, even if I only add a single small change to existing file and even if I only add a new ipynb file that is independent of the previous ones. When some types of contents that are fine in vscode jupyter notebook fails to compile in quarto, I spend hours on debugging this. 

`quarto preview` can be faster since it watch file changes. However, I don't know where to find the files rendered by `quarto preview`. After all, I want to publish my docs, so I need the files, not the just the web page showing in my browser. 

It seems Documentations https://quarto-dev.github.io/quarto-r/reference/quarto_preview.html and https://quarto.org/docs/tools/text-editors.html didn't cover this, so I came here to ask a question.

**Comment by 2catycm** [comment]
I've seen this discussion https://github.com/quarto-dev/quarto-cli/discussions/4684

**Comment by mcanouil** [comment]
The output are located either next to the Quarto document used as input or inside the `output-dir` if you set one.
The behaviour is the same with `quarto render` and `quarto preview`.
`quarto preview` is `quarto render` + watching input files + opening the rendered files.

Note that the R package is only a wrapper and is not the main tool.
Use `quarto --help` to get Quarto CLI helps.

> didn't cover this, so I came here to ask a question.

How so? From the link you shared I can read:

> The document will be rendered and a web browser with a “live preview” opened.

And there is a "render without preview" section: https://quarto.org/docs/tools/text-editors.html#render-without-preview

What's not clear or missing?

**Comment by mcanouil** [comment]
Regarding you project taking time, are you leveraging caching/freezing?

- https://quarto.org/docs/projects/code-execution.html

Incremental rendering?

Don't `quarto render/preview`, do `quarto render/preview path/to/my/new/file.qmd/ipynb`.
